miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryController.java

```diff
@@ -906,30 +785,75 @@ public ModelAndView editBulkLibraries(@PathVariable String libraryIds, ModelMap
           throw new IOException("Can only update libraries when samples all have the same class.");
         }
       }
-      model.put("title", "Bulk Edit Libraries");
-      model.put("librariesJSON", mapper.writerFor(new TypeReference<List<LibraryDto>>() {
-      }).writeValueAsString(libraryDtos));
-      model.put("method", "Edit");
+      return results;
+    }
 
-      JSONArray libraryDesigns = new JSONArray();
-      if (sampleClass != null) {
-        libraryDesigns.addAll(requestManager.listLibraryDesignByClass(sampleClass));
-      }
-      model.put("libraryDesignsJSON", libraryDesigns.toString());
-      JSONArray libraryDesignCodes = new JSONArray();
-      libraryDesignCodes.addAll(requestManager.listLibraryDesignCodes());
-      model.put("libraryDesignCodesJSON", libraryDesignCodes.toString());
-      model.put("platformTypes", mapper.writeValueAsString(populatePlatformTypes(currentPlatforms)));
+    @Override
+    protected LibraryDto asDto(Library model) {
+      return Dtos.asDto(model);
+    }
 
-      return new ModelAndView("/pages/bulkEditLibraries.jsp", model);
-    } catch (IOException ex) {
-      if (log.isDebugEnabled()) {
-        log.error("Failed to get bulk libraries", ex);
-      }
-      throw ex;
+    @Override
+    protected Class<? extends LibraryDto> getDtoClass() {
+      return isDetailedSampleEnabled() ? DetailedLibraryDto.class : LibraryDto.class;
     }
+
+    @Override
+    protected void writeConfiguration(ObjectMapper mapper, ObjectNode config) {
+      config.put("showDescription", showDescription);
+      config.put("showVolume", showVolume);
+      config.put("showLibraryAlias", showLibraryAlias);
+    }
+  };
+
+  /**
+   * used to edit samples with ids from given {sampleIds} sends Dtos objects which will then be used for editing in grid
```

Comment:
I'm a bit confused about this documentation -- are these indeed samples being edited? And is this method able to handle both edit and propagate?